code/API_definitions/qod-api.yaml

```diff
@@ -290,6 +290,67 @@ paths:
         - threeLegged:
             - "qod-sessions-delete"
 
+  /sessions/{sessionId}/extend:
+    post:
+      tags:
+        - Extend or reduce QoS session duration
+      summary: "Extend or reduce QoS session duration"
+      description: |
+        Extend or reduce the duration of an active QoS session.
+        If this operation is executed successfully, the new duration of the target session will be original duration plus extended duration.
+        If the new duration is shorter than or equal to elapsed duration, the session will be terminated.
+      operationId: extendQosSessionDuration
```

Comment:
As you can use operation to increase/decrease duration I think we can name operation changeQosSessionDuration.

Reply:
Well, I do have a theoretical use case in mind but I would like to validate with the community.

Let's say a teleoperator uses a QoD session to boost the connection for remotely controlling a car. 
As the initial ETA (Estimated Time of Arrival) of the trip is one hour, the teleoperating application server creates a session with duration 60 minutes. Along the trip, since the traffic situation is better than expected, the ETA is adjusted to 45 minutes, the teleoperating application server may reduce the session duration to 45 minutes.

Reply:
> Well, I do have a theoretical use case in mind but I would like to validate it with the community.
> 
> Let's say a teleoperator uses a QoD session to boost the connection for remotely controlling a car. As the initial ETA (Estimated Time of Arrival) of the trip is one hour, the teleoperating application server creates a session with duration 60 minutes. Along the trip, since the traffic situation is better than expected, the ETA is adjusted to 45 minutes, the teleoperating application server may reduce the session duration to 45 minutes.

@emil-cheung well, I would consider this use case also theoretically :-) In the use case it is much more pratical to `DELETE` the session explicitly when the car arrived at the destination and/or no longer needs the quality boost.But the other way round - extending the session duration if the journey takes longer then expected - is for sure an important use case.

My proposal is to limit the scope of the issue and the PR to extending the session duration and omit "reduce" from it (I have seen that the issue is already only mentioning extension).

Reply:
I am also in favour of only allowing extensions to the session duration, and not reductions.

An additional point. If session duration reduction is allowed, then a reduction that would reduce the session duration to zero or below should be considered an error, rather than just deleting the session. There are two reasons for this:
- it probably is an error, because if the API consumer had actually wanted to delete the session, they could do that explicitly
- the required scope for this endpoint would probably be `qod:sessions:update` according to [this](https://github.com/camaraproject/Commonalities/pull/57) PR in Commonalties, and we would then be allowing a `delete` with only `update` scope

Reply:
As several companies prefer starting with only duration extension, I have updated both the issue and the PR to duration extension only. If in the future, there is need for duration reduction, let's open a new issue to improve.

Reply:
Are we allowing remaining + extension > maximum session duration. If not we should add this a constraint in the description and consider an error for that.

Reply:
@jlurien has a good point.  This should not be a workaround for the maximum session duration.

Would we even further constrain this so that initial duration + extension < max session duration?

Reply:
Within last QoD call we have discussed to constrain in such way, that the resulting new remaining session duration should not extend the maximum session duration (of 86400 seconds).

So: 
IF remaining + extension <= maximum session duration, duration will be set to remaining duration + extension

We also discussed the case if remaining duration + extension > maximum session duration in the extension request. In this case there was the notion that we should not raise an error, but set the session duration to the value where the new remaining duration is equal to the maximum session duration (as the developer can't calculate the accurate maximum extension and there would be several race conditions possible).

The all cases the actual new duration will be in the response.

Reply:
I would keep just the extension, not reduction, so operation name is valid